### PR TITLE
Add stack diagram to Ceramic advanced overview document - #PLAT-390

### PR DESCRIPTION
**Pull request by gvelez17**
#Add stack diagram to Ceramic advanced overview document - #PLAT-390

## Description

Include a node component diagram in the overview document for greater clarity.

## How Has This Been Tested?

Describe the tests that you ran to verify your changes. Provide instructions for reproduction.

- [ ] Test A (e.g. Test A - New test that ... ran in local, docker, and dev unstable.)
- [ ] Test B

## Definition of Done

Before submitting this PR, please make sure:

- [ ] The work addresses the description and outcomes in the issue
- [ ] I have added relevant tests for new or updated functionality
- [x ] My code follows conventions, is well commented, and easy to understand
- [ ] My code builds and tests pass without any errors or warnings
- [ ] I have tagged the relevant reviewers
- [ ] I have updated the READMEs of affected packages
- [x ] I have made corresponding changes to the documentation
- [ ] The changes have been communicated to interested parties

## References:

Please list relevant documentation (e.g. tech specs, articles, related work etc.) relevant to this change, and note if the documentation has been updated.

pulled from https://github.com/3box/ceramic-infra/tree/dev/docs

### CODE REVIEW

**Review by oed — COMMENTED** [review]
Hm, I think calling out that a Ceramic node needs to talk to a CAS node makes sense in this document. If no ipfs node config is provided Ceramic will actually start a go-ipfs node automatically, so tha part is not needed here. Also the use of S3 is specific to our deployments, not something that is true for any Ceramic deployment.

File: docs/learn/advanced/overview.md [hunk 1]
```diff
@@ -77,6 +77,12 @@ In addition to executing stream transactions according to StreamType logic, Cera
 - **Query responses:** Ceramic nodes respond to stream queries from clients. If the node has the stream pinned, it will return the response; if not, it will ask the rest of the network for the stream over libp2p and then return the response.
 - **Broadcasting transactions:** When a Ceramic node successfully performs a transaction on a stream, it broadcasts this transaction out the rest of the network over libp2p so other nodes also pinning this stream can update their StreamState to reflect this new transaction.
 
+### Components of a Ceramic Node
+
+A ceramic node requires several components in addition to the actual ceramic instance: an IPFS running instance and storages, a ceramic state store, and log storage.
```
Comment (v-stickykeys) [inline docs/learn/advanced/overview.md:82]: This is a bit confusing. Who is the intended audience and what level of abstraction are we aiming to give?

Is this information for node operators? Or is it to help people understand how the Ceramic system works?

If the latter than it should just be 3 components of the system:
- ceramic daemon
- ipfs
- ceramic anchor service

Comment (v-stickykeys) [reply docs/learn/advanced/overview.md:82]: Also diagram should reflect the abstraction accurately. This seems like a diagram for node operators and I think would be more confusing than helpful for general public--i.e. people will think Ceramic *relies on* S3

**Review by v-stickykeys — CHANGES_REQUESTED** [review]
Need to understand intended audience than update this to meet what they need to understand.

The paragraph doesn't describe what's on the diagram yet.

**Comment by gvelez17** [comment]
Simplified the diagram and removed any explicit mention of S3 - does this address comments above @oed @v-stickykeys ?

**Review by oed — APPROVED** [review]
This is ok for now, but I imagine these things will change over time so we should keep this page in mind when that happens.

**Comment by gvelez17** [comment]
> This is ok for now, but I imagine these things will change over time so we should keep this page in mind when that happens.

Docs always rot.  

Unfort diagrams are useful and can't easily be autogenerated :-)

**Review by v-stickykeys — APPROVED** [review]
Nice! Much clearer now

**Pull request closed without merging**